modules/util/commit_trailers.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package util
+
+import (
+	"errors"
+	"net/mail"
+	"strings"
+)
+
+var ErrInvalidCommitTrailerValueSyntax = errors.New("syntax error occurred while parsing a commit trailer value")
+
+// ParseCommitTrailerValueWithAuthor parses a commit trailer value that contains author data.
+// Note that it only parses the value and does not consider the trailer key i.e. we just
+// parse something like the following:
+//
+// Foo Bar <foobar@example.com>
```

Comment:
Looks like it's a standard format, why not use `mail.ParseAddress` directly?

Reply:
I think I originally was worried about using this since I wasn't too sure of what `mail.ParseAddress` accepted (i.e. wasn't familiar with [RFC 5322](https://www.rfc-editor.org/rfc/rfc5322.html)) as input but looking at it now it seems to be a good solution compared to having us parsing the input manually.

It may still make sense to have a utility function around since it looks like we can accept a trailer with no name e.g.:
```go
addrStr := "<hello@world>"
addr, _ := mail.ParseAddress(addrStr)
fmt.Printf("%s", addr.Address)
// hello@world
// Program exited.
```

Here we can throw an error if `addr.Name` is empty.